app/utils/deep_link/index.ts

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createIntl, IntlShape} from 'react-intl';
+import urlParse from 'url-parse';
+
+import {makeDirectChannel, switchToChannelByName} from '@actions/remote/channel';
+import {appEntry} from '@actions/remote/entry';
+import {showPermalink} from '@actions/remote/permalink';
+import {fetchUsersByUsernames} from '@actions/remote/user';
+import {DeepLink, Launch, Screens} from '@constants';
+import {getDefaultThemeByAppearance} from '@context/theme';
+import DatabaseManager from '@database/manager';
+import {DEFAULT_LOCALE, getTranslations} from '@i18n';
+import {getActiveServerUrl} from '@queries/app/servers';
+import {getCurrentUser, queryUsersByUsername} from '@queries/servers/user';
+import {dismissAllModalsAndPopToRoot, showModal} from '@screens/navigation';
+import EphemeralStore from '@store/ephemeral_store';
+import NavigationStore from '@store/navigation_store';
+import {errorBadChannel, errorUnkownUser} from '@utils/draft';
+import {logError} from '@utils/log';
+import {escapeRegex} from '@utils/markdown';
+import {addNewServer} from '@utils/server';
+import {removeProtocol} from '@utils/url';
+
+import type {DeepLinkChannel, DeepLinkDM, DeepLinkGM, DeepLinkPermalink, DeepLinkPlugin, DeepLinkWithData, LaunchProps} from '@typings/launch';
+
+export async function handleDeepLink(deepLinkUrl: string, intlShape?: IntlShape, location?: string) {
+    try {
+        const parsed = parseDeepLink(deepLinkUrl);
+        if (parsed.type === DeepLink.Invalid || !parsed.data || !parsed.data.serverUrl) {
+            return {error: true};
+        }
+
+        const currentServerUrl = await getActiveServerUrl();
```

Comment:
why would `currentServerUrl` be undefined here ?

Reply:
https://github.com/mattermost/mattermost-mobile/pull/6869#discussion_r1049537589